pkgs/tools/misc/broot/default.nix

```diff
@@ -53,6 +53,7 @@ rustPlatform.buildRustPackage rec {
     wrapProgram $out/bin/broot \
       --set BR_INSTALL no
 
+  '' + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
     # Install shell function for bash.
     $out/bin/broot --print-shell-function bash > br.bash
```

Comment:
You can use `stdenv.hostPlatform.emulator buildPackages` to run the binary with qemu-user, and keep cross compiled packages feature on par with native ones.

Reply:
`${lib.optionalString (stdenv.hostPlatform.emulatorAvailable buildPackages) (stdenv.hostPlatform.emulator buildPackages)}`

Reply:
```diff
diff --git a/pkgs/tools/misc/broot/default.nix b/pkgs/tools/misc/broot/default.nix
index 646ab244682..b068f5cfa5c 100644
--- a/pkgs/tools/misc/broot/default.nix
+++ b/pkgs/tools/misc/broot/default.nix
@@ -11,6 +11,7 @@
 , Security
 , xorg
 , zlib
+, buildPackages
 }:
 
 rustPlatform.buildRustPackage rec {
@@ -53,9 +54,9 @@ rustPlatform.buildRustPackage rec {
     wrapProgram $out/bin/broot \
       --set BR_INSTALL no
 
-  '' + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+  '' + lib.optionalString (stdenv.hostPlatform.emulatorAvailable buildPackages) ''
     # Install shell function for bash.
-    $out/bin/broot --print-shell-function bash > br.bash
+    ${stdenv.hostPlatform.emulator buildPackages} $out/bin/broot --print-shell-function bash > br.bash
     install -Dm0444 -t $out/etc/profile.d br.bash
 
     # Install shell function for zsh.
```

this fails without error message :(

Reply:
This fails without a error message

```diff
diff --git a/pkgs/tools/misc/broot/default.nix b/pkgs/tools/misc/broot/default.nix
index 646ab244682..6a93833bc49 100644
--- a/pkgs/tools/misc/broot/default.nix
+++ b/pkgs/tools/misc/broot/default.nix
@@ -11,6 +11,7 @@
 , Security
 , xorg
 , zlib
+, buildPackages
 }:
 
 rustPlatform.buildRustPackage rec {
@@ -53,17 +54,17 @@ rustPlatform.buildRustPackage rec {
     wrapProgram $out/bin/broot \
       --set BR_INSTALL no
 
-  '' + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+  '' + lib.optionalString (stdenv.hostPlatform.emulatorAvailable buildPackages) ''
     # Install shell function for bash.
-    $out/bin/broot --print-shell-function bash > br.bash
+    ${stdenv.hostPlatform.emulator buildPackages} $out/bin/broot --print-shell-function bash > br.bash
     install -Dm0444 -t $out/etc/profile.d br.bash
 
     # Install shell function for zsh.
-    $out/bin/broot --print-shell-function zsh > br.zsh
+    ${stdenv.hostPlatform.emulator buildPackages} $out/bin/broot --print-shell-function zsh > br.zsh
     install -Dm0444 br.zsh $out/share/zsh/site-functions/br
 
     # Install shell function for fish
-    $out/bin/broot --print-shell-function fish > br.fish
+    ${stdenv.hostPlatform.emulator buildPackages} $out/bin/broot --print-shell-function fish > br.fish
     install -Dm0444 -t $out/share/fish/vendor_functions.d br.fish
 
   '' + ''
```

Reply:
Oh that is a bash wrapper, personally don't know how to run anything with a shebang with qemu-user.

Reply:
Luckily we have makeBinaryWrapper

Reply:
doesn't work with `makeBinaryWrapper` either so i moved the wrapProgram to after these